src/graphs/SMG.cc

```diff
@@ -38,6 +39,16 @@ const std::map<SMGValue, SMGEdgePointsToPtr>& SMG::GetPTEdges() const { return p
 
 const SMGEntitySet<const SMGEdgeHasValue>& SMG::GetHVEdges() const { return hv_edges_; }
 
+const SMGEntitySet<const SMGEdgeHasValue> SMG::GetHVEdges(
+    const SMGEdgeHasValueFilter& filter) const {
+  SMGEntitySet<const SMGEdgeHasValue> to_return;
+  std::copy_if(hv_edges_.begin(),
+               hv_edges_.end(),
+               std::inserter(to_return.set(), to_return.begin()),
+               filter);
+  return to_return;
+}
```

Comment:
That is obvious and not what I am addressing here. IMHO:
A) It should just call FilterSet(hv_edges_&), compiler might even optimize out the extra function call. 
B) aside from one function is using class field and the second function argument, they do the exactly same thing - but one with std::copy_if and the second with for-loop. Which is better? And - that should be unified.